meddlr/metrics/ssfd.py

```diff
@@ -0,0 +1,260 @@
+from typing import Sequence
+
+import torch
+from torch import nn
+
+from meddlr.metrics.functional.image import mse
+from meddlr.metrics.metric import Metric
+from meddlr.ops import complex as cplx
+from meddlr.utils import env
+
+# Pre-trained model for computing SSFD. This is a UNet model trained on a self-supervsed
+# in-painting pretext task on the fastMRI knee dataset.
+SSFD_HUGGINGFACE_URL = "https://huggingface.co/philadamson93/SSFD/resolve/main/model.ckpt"
+
+
+class SSFD(Metric):
+    """
+    Self-Supervised Feature Distance. SSFD evaluates the feature distance between a
+    pair of images from features extracted from a pre-trained neural network trained on
+    a self-supervision task with fastMRI datasets [1]. SSFD has been shown to correspond
+    well to Radiologist Reader Scores of accelerated  MR reconstructions.
+
+    References:
+    ..  [1] Adamson, Philip M., et al.
+        SSFD: Self-Supervised Feature Distance as an MR Image Reconstruction Quality Metric."
+        NeurIPS 2021 Workshop on Deep Learning and Inverse Problems. 2021.
+        https://openreview.net/forum?id=dgMvTzf6M_3
+    """
+
+    def __init__(
+        self,
+        mode: str = "grayscale",
+        layer_names: Sequence[str] = ["block4_relu2"],
```

Comment:
All default inputs should be immutable - this should be a tuple instead of a list by default